performance/src/efficiency/finding_performance_writer.cpp

```diff
@@ -257,6 +257,7 @@ void finding_performance_writer::write_common(
             ptc.vertex[2] < m_cfg.truth_config.z_min ||
             ptc.vertex[2] > m_cfg.truth_config.z_max ||
             vector::perp(ptc.vertex) > m_cfg.truth_config.r_max ||
+            ptc.process != m_cfg.truth_config.is_HS ||
```

Comment:
The problem with this is that it will permanently affect all of our truth filtering, even if we do not want to select only the HS vertex.

To make this more robust, I'd suggest making this `bool is_HS` into a `int process_id`, and then we can use the logic than any negative value (default) selects _any_ process, and a positive ID selects only that process:


```suggestion
            (m_cfg.truth_config.process_id >= 0 && m_cfg.truth_config.process_id == ptc.process)
```